whois-commons/src/main/java/net/ripe/db/whois/common/conversion/PasswordFilter.java

```diff
@@ -20,15 +21,26 @@ public class PasswordFilter {
 
     //from logsearch tweaked
     private static final Pattern PASSWORD_PATTERN_FOR_CONTENT = Pattern.compile("(?im)^(override|password)(:|%3A)\\s*(.+)\\s*$");
+    private static final Pattern BASIC_AUTH_HEADER_PATTERN_FOR_CONTENT = Pattern.compile("(?im)^Header: Authorization=Basic *([^ ]+) *$", Pattern.CASE_INSENSITIVE);
+
     private static final Pattern URI_PASSWORD_PATTERN_PASSWORD_FOR_URL = Pattern.compile("(?<=)(password|override)(:|=|%3A)([^&^\\s]*)", Pattern.CASE_INSENSITIVE);
 
     public static String filterPasswordsInContents(final String contents) {
-        String result = contents;
-        if (contents != null) {
-            final Matcher matcher = PASSWORD_PATTERN_FOR_CONTENT.matcher(contents);
-            result = replacePassword(matcher);
+        if(StringUtils.isEmpty(contents)) {
+            return contents;
         }
-        return result;
+
+        final String filteredContent = replaceBasicAuthHeader(BASIC_AUTH_HEADER_PATTERN_FOR_CONTENT.matcher(contents));
+        return replacePassword(PASSWORD_PATTERN_FOR_CONTENT.matcher(filteredContent));
+    }
+
+    private static String replaceBasicAuthHeader(final Matcher matcher)  {
+        final StringBuilder result = new StringBuilder();
+        while (matcher.find()) {
+            matcher.appendReplacement(result, String.format("%s%s", StringUtils.substringBefore(matcher.group(0), matcher.group(1)),"FILTERED"));
```

Comment:
The regex is still matching a space? Can you use `(.*)` as the group in the pattern instead?

Reply:
I tried to simplify the regex @maggarwal13 please review

Reply:
Sonar is complaining , else I am fine

Reply:
It was CodeQL code scanning, I dismissed the alert.